### ISSUE TITLE
"enable-sem": true，root user can't execute select @@tidb_config, but can download tidb config from dashboard

### ISSUE THREAD

**Issue by seiya-annie** [pingcap/tidb#25425]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

1. set "enable-sem": true
2. select @@tidb_config;
3. download from dashboard 

### 2. What did you expect to see? (Required)
the behavior in dashboard and MySQL client should be same;

### 3. What did you see instead (Required)
root user can download tidb config from dashboard

### 4. What is your TiDB version? (Required)

| Release Version: v5.1.0-20210611
Edition: Community
Git Commit Hash: 29bfbb2
Git Branch: heads/refs/tags/v5.1.0-20210611
UTC Build Time: 2021-06-11 07:52:09
GoVersion: go1.16.4
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false |

**Comment by morgo** [comment]
I believe the best way to solve this is to create a dynamic privilege like `DASHBOARD_ADMIN` and create an internal TiDB Dashboard user. That will allow it to operate with a credential set that makes the most sense to us. @breeswish what do you think?

**Comment by breeswish** [comment]
> I believe the best way to solve this is to create a dynamic privilege like `DASHBOARD_ADMIN` and create an internal TiDB Dashboard user. That will allow it to operate with a credential set that makes the most sense to us. @breeswish what do you think?

Yes there is a plan to add Dashboard privileges, but it might be unrelated with this issue. After deeper discussions with @bb7133 I think what we met is that, the TiDB:10080 `/config` endpoint provides all config items by design, which reveals some hidden config items in the global variable. Users can download the content of `/config` endpoint (and also other endpoints) via TiDB Dashboard UI in the Advanced Debugging feature. 

Considering that all these things are debug information (and in UI there is even a notice for it, see below), revealing internal or hidden things (or even things that user cannot understand at all) should be expected as by design. What do you think? @morgo @seiya-annie 

![image](https://user-images.githubusercontent.com/1916485/122157910-486a1b00-ce9e-11eb-9b40-1212181e6a57.png)

**Comment by morgo** [comment]
That is a question I'd have to run by @SunRunAway who is the PM :-)

As an alternative, you should be able to see that SEM is enabled in this variable list witb `tidb_enable_enhanced_security=ON`. In which case, you could disable the complete use of this feature.

**Comment by seiya-annie** [comment]
ok

**Comment by breeswish** [comment]
> That is a question I'd have to run by @SunRunAway who is the PM :-)
> 
> As an alternative, you should be able to see that SEM is enabled in this variable list witb `tidb_enable_enhanced_security=ON`. In which case, you could disable the complete use of this feature.

I would supply some background information :) For TiDB Cloud use cases where SEM is on, this feature is not available for users, but is available for us, which seems to be at a nice balance. @morgo

**Comment by morgo** [comment]
@SunRunAway can you describe the expected behavior?

**Comment by SunRunAway** [comment]
Chatted with @breeswish, he will provide a document for describing privileges required by each component in TiDB-dashboard.
